src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -227,12 +227,78 @@ public ISqlConnectionFactory ConnectionFactory
         // Attempts to link a URI to an actively used connection for this URI
         public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connectionInfo) => this.OwnerToConnectionMap.TryGetValue(ownerUri, out connectionInfo);
 
+        /// <summary>
+        /// Refreshes the auth token of a given connection 
+        /// </summary>
+        /// <param name="ownerUri">The URI of the connection</param>
+        /// <returns>true upon successful refresh of the auth token,
+        /// false if the uri cannot be found</returns>
+
+        internal async Task<bool> TryRefreshAuthToken(string ownerUri)
+        {
+            ConnectionInfo connInfo;
+            if (this.TryFindConnection(ownerUri, out connInfo))
+            {
+                // If not an azure connection, no need to refresh token
+                if (connInfo.ConnectionDetails.AuthenticationType != "AzureMFA") 
+                {
+                    return false;
+                } 
+                else 
+                {
+                    // Check if token is expired
+                    var maxTolerance = 2 * 60; // two minutes
+                    if (connInfo.ConnectionDetails.ExpiresOn - DateTimeOffset.Now.ToUnixTimeSeconds() < maxTolerance)
+                    {   
+                        // disable intellisense
+                    } 
+                    else 
+                    {
+                        return false;
+                    }
+                }
+            }
+            if (!this.TryFindConnection(ownerUri, out ConnectionInfo connection))

```

Comment:
what do we have to do the check again? this seems can be the else block of the previous if statement.